scripts/o3de/o3de/disable_gem.py

```diff
@@ -98,12 +98,66 @@ def disable_gem_in_project(gem_name: str = None,
     return ret_val
 
 
+def remove_explicit_gem_activation_for_all_paths(gem_root_folders: list,
+                                 project_name: str = None,
+                                 project_path: pathlib.Path = None,
+                                 enabled_gem_file: pathlib.Path = None) -> int:
+    """
+    walks each gem root folder directory structure and removes explicit
+    activation of the gems to the project
+    :param gem_root_folders: name of the gem to add
+    :param project_name: name of to the project to add the gem to
+    :param project_path: path to the project to add the gem to
+    :param enabled_gem_file: if this dependency goes/is in a specific file
+    :return: 0 for success or non 0 failure code
+    """
+    if not gem_root_folders:
+        logger.error('gem_root_folders list cannot be empty')
+        return 1
+
+    # Skip removing of explicit activation of gems within template directories
+    # They shouldn't be activated anyway
+    def stop_on_template_folders(directories: list, filenames: list) -> bool:
+        return 'template.json' in filenames
+
+    gem_dirs_set = set()
+    ret_val = 0
+    for gem_root_folder in gem_root_folders:
+        gem_root_folder = pathlib.Path(gem_root_folder).resolve()
+        if not gem_root_folder.is_dir():
+            logger.error(f'gem root folder of {gem_root_folder} is not a directory')
+            ret_val = 1
+        for root, dirs, files in os.walk(gem_root_folder):
+            # Skip template directories
+            if stop_on_template_folders(dirs, files):
+                dirs[:] = []
+            if 'gem.json' in files:
```

Comment:
```suggestion
            elif 'gem.json' in files:
```

Reply:
not sure I fully understand how `stop_on_template_folders` prevents the check if `gem.json` in files here.  I suppose you aren't going to have a gem.json file in the same folder as a template.json so it probably doesn't really matter.

Reply:
I'll make that an else if to prevent the situation where a `template.json` happens to be in the same folder as a `gem.json`.

We would consider that a "template" in that case instead of a "gem>